src/core/stompest/protocol/frame.py

```diff
@@ -82,9 +83,13 @@ def __repr__(self):
 
     def __str__(self):
         """Render the wire-level representation of a STOMP frame."""
-        headers = sorted(self.headers.iteritems()) if self.rawHeaders is None else self.rawHeaders
-        headers = ''.join('%s:%s%s' % (self._encode(self._escape(unicode(key))), self._encode(self._escape(unicode(value))), StompSpec.LINE_DELIMITER) for (key, value) in headers)
-        return StompSpec.LINE_DELIMITER.join([self._encode(unicode(self.command)), headers, '%s%s' % (self.body, StompSpec.FRAME_DELIMITER)])
+        headers = sorted(self.headers.items()) if self.rawHeaders is None else self.rawHeaders
```

Comment:
Please factor out the distinction of cases s.t. no redundant code remains, e.g. `cast = unicode if isPython3() else str`).